src/Main/ProjectFile/ProjectFileLoader.cpp

```diff
@@ -82,6 +82,15 @@ void ProjectFileLoader::Load(const QString &filename) {
     }
   }
 
+  // set device
```

Comment:
How's this move solving the behavior reported on the ticket ? Like why is it not set to None now.

Reply:
This code causes save file during loading. Just before task state loading. It is very complicated and better to have a call for explanation. But I put additional protection against this.